cpp/ParseFasm.cpp

```diff
@@ -0,0 +1,414 @@
+#include "antlr4-runtime.h"
+#include "FasmLexer.h"
+#include "FasmParser.h"
+#include "FasmParserVisitor.h"
+
+// External C Interface
+//
+// These functions serialize the FASM parse tree to an easy to parse
+// tag/length/value binary format, where the tag is one byte and
+// the length is 4 bytes, in native endianness (typically little.)
+extern "C" {
+  const char *from_string(const char *in, bool hex);
+  const char *from_file(const char *path, bool hex);
+}
+
+using namespace antlr4;
+using namespace antlrcpp;
+
+// Hex mode is useful for debugging.
+// In this mode, binary values are printed as hex values surrounded by < >
+bool hex_mode = false;
```

Comment:
Should this be an argument to the C interface rather than a global constants?

Reply:
It's noisy to thread this through everything, and it's only a debug feature, like debug levels.